core/src/main/java/com/opensymphony/xwork2/util/DebugUtils.java

```diff
@@ -37,4 +42,12 @@ public static void notifyDeveloperOfError(Logger log, Object action, String mess
         }
     }
 
+    /**
+     * @since 7.0
+     */
+    public static void logWarningForFirstOccurrence(String key, Logger log, String msg, Object... args) {
```

Comment:
This method isn't really a `DebugUtils` method but couldn't think of a better home - can create a separate class if you prefer